core/src/conn.rs

```diff
@@ -1,10 +1,92 @@
 use crate::odbc_uri::UserOptions;
 use crate::{err::Result, Error};
 use crate::{MongoQuery, TypeMode};
-use bson::{doc, Bson, UuidRepresentation};
-use mongodb::Client;
+use lazy_static::lazy_static;
+use mongodb::options::AuthMechanism;
+use mongodb::{
+    bson::{doc, Bson, UuidRepresentation},
+    Client,
+};
 use serde::{Deserialize, Serialize};
-use std::time::Duration;
+#[cfg(feature = "garbage_collect")]
+use std::sync::Weak;
+use std::{sync::Arc, time::Duration};
+use tokio::runtime::Runtime;
+
+// we make from UserOptions to Client and Weak<Runtime> so that we do not hold around
+// Clients and Runtimes that are no longer in use. In most cases it won't matter, but for drivers that live in
+// memory for a long time, this could be a problem.
+#[cfg(feature = "garbage_collect")]
```

Comment:
garbage_collect is expected to only be used by users building their own tools with our driver that won't unload the driver all the time :joy: